sktime/classification/deep_learning/cnn.py

```diff
@@ -0,0 +1,143 @@
+    https://github.com/hfawaz/dl-4-tsc/blob/master/classifiers/cnn.py
+    """
+
+    def __init__(
+        self,
+        n_epochs=2000,
+        batch_size=16,
+        kernel_size=7,
+        avg_pool_size=3,
+        n_conv_layers=2,
+        callbacks=None,
+        verbose=False,
+    ):
+        super(CNNClassifier, self).__init__()
+        self.n_conv_layers = n_conv_layers
+        self.avg_pool_size = avg_pool_size
+        self.kernel_size = kernel_size
+        self.callbacks = callbacks
+        self.n_epochs = n_epochs
+        self.batch_size = batch_size
+        self.verbose = verbose
+
+    def build_model(self, input_shape, n_classes, **kwargs):
+        """Construct a compiled, un-trained, keras model that is ready for training.
+
+        Parameters
+        ----------
+        input_shape : tuple
+            The shape of the data fed into the input layer, should be (m,d)
+        n_classes: int
+            The number of classes, which becomes the size of the output layer
+
+        Returns
+        -------
+        output : a compiled Keras Model
+        """
+        from tensorflow import keras
+
+        input_layer, output_layer = self.build_network(input_shape, **kwargs)
+
+        output_layer = keras.layers.Dense(units=n_classes, activation="sigmoid")(
+            output_layer
+        )
+
+        model = keras.models.Model(inputs=input_layer, outputs=output_layer)
+        model.compile(
+            loss="mean_squared_error",
+            optimizer=keras.optimizers.Adam(),
+            metrics=["accuracy"],
+        )
+        return model
+
+    def _fit(self, X, y, **kwargs):
+        """Fit the classifier on the training set (X, y).
+
+        Parameters
+        ----------
+        X : np.ndarray of shape = (n_instances, n_dimensions, series_length)
+            The training input samples.
+        y : np.ndarray of shape = [n_instances]
+            The training data class labels.
+        input_checks : boolean
+            whether to check the X and y parameters
+        validation_X : np.ndarray of shape = (n_instances, n_dimensions, series_length)
+            The validation samples.
+            Unless strictly defined by the user via callbacks (such as
+            EarlyStopping), the presence or state of the validation
+            data does not alter training in any way. Predictions at each epoch
+            are stored in the model's fit history.
+        validation_y : np.ndarray of shape = [n_instances]
+            The validation class labels.
+
+        Returns
+        -------
+        self : object
+        """
+        if self.callbacks is None:
+            self._callbacks = []
+        y_onehot = self.convert_y(y)
+        check_random_state(self.random_state)
+
+        # Transpose to conform to Keras input style.
+        X = X.transpose(0, 2, 1)
+        self.input_shape = X.shape[1:]
+        self.model_ = self.build_model(self.input_shape, self.n_classes_)
```

Comment:
this looks like much of it will be the same - should this be boilerplate that is in the intermediate base class?
Can you perhaps get rid of `_fit` entirely?

Reply:
possibly, I'll look through on the next PR to see if it could be moved up to the base.